platform/default/src/mbgl/storage/database_file_source.cpp

```diff
@@ -179,6 +179,8 @@ std::unique_ptr<AsyncRequest> DatabaseFileSource::request(const Resource& resour
 }
 
 void DatabaseFileSource::forward(const Resource& res, const Response& response, std::function<void()> callback) {
+    if (res.storagePolicy == Resource::StoragePolicy::Volatile) return;
```

Comment:
We should also have something like this when requesting volatile resources from the offline database. If they are never stored there is no point in querying for them.